### DISCUSSION TITLE
ERROR: An internal database error occurred: disk I/O error.

### DISCUSSION THREAD

**Discussion by BKSVitor** [abraunegg/onedrive#2677] [Q&A] [answered]
Hello everyone, I tried the suggestions that abraunegg reported, however, the return is still the same. I need to share information that may or may not be related. 

Info of my case: https://github.com/abraunegg/onedrive/issues/2675

abraunegg's suggestion: https://github.com/abraunegg/onedrive/wiki/Frequently-Asked-Questions#an-internal-database-error-occurred-disk-io-error

The Microsoft Onedrive is having issues on last days, I share some folders via link that are in the cloud using OneDrive, a few days ago I noticed a problem with downloading files, before I was not asked for a password to download files and now even with the email account and password I have problems doing the file download. I will forward a gif of the problem related to OneDrive, I have opened a ticket with Microsoft support and I'm waiting for a response.

In the video, this is the scenario:

https://github.com/abraunegg/onedrive/assets/142534915/0e8f4361-dcb0-464f-88de-fd8712a5a6bf

1- I receive the link and password to access folders in the cloud;
2- I select the file I want to download (a few days ago it was not necessary to enter any type of account or login)
3- After entering my credentials with email and password, I receive the error after several redirects.

Sorry for my english, Portuguese is my mother language.

**Comment by abraunegg** [comment, marked as answer]
@BKSVitor 
The 'browser' redirect has nothing to do with the 'onedrive' client.

My suggestion here is that you clear all your browser data , clear any browser stored passwords for Microsoft OneDrive and try those links again.

Your issue in https://github.com/abraunegg/onedrive/issues/2675 is that you are running a systemd service .. potentially 2 of them at the same time, or running 1 systemd service and then trying to run the client manually - you cannot do this. Those actions result in the error you are getting (An internal database error occurred: disk I/O error).

### Actions for you to solve
1. Search your systemd services for all 'onedrive' related items. Refer to your Linux distribution forums for assistance if you do not know how to do this
2. Disable and remove all of the systemd services for 'onedrive' that you find
3. Run the client 'manually' - `onedrive --synchronize --verbose` and confirm that you do not get your disk I/O error
4. Go back and re-read https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-in-monitor-mode and only enable 1 systemd service

In future, if you then want to manually run the client (step 3 above) you *must* first *stop* the systemd service you have enabled in step 4 otherwise you will again get the error you are facing.

**Reply by BKSVitor** [reply]
Abraunegg, I'm sorry for the delay in responding to you, this week I had some problems that prevented me from continuing my report.

I will follow your suggestions and get back to you soon with the results! Thanks a lot for the help!

**Reply by BKSVitor** [reply]
Abraunegg, sorry for taking so long to answer.

your suggestions work well, to help with sync automation I made a bash script and scheduled it with crontab.

```
#!/bin/bash

# Stop Sync
systemctl --user stop onedrive

# Execute resync
echo "Y" | onedrive --resync

# Start Sync
onedrive destination-directory /media/casmed/onedrive/ --synchronize
```

**Reply by abraunegg** [reply]
@BKSVitor 
Please read the documentation ... your bash script is plagued by errors:
```
#!/bin/bash

# Stop Sync
systemctl --user stop onedrive

# Execute resync
echo "Y" | onedrive --resync     <---- This is the 100% incorrect way to do this. Please read the documentation

# Start Sync
onedrive destination-directory /media/casmed/onedrive/ --synchronize   <---- You are referencing 100% incorrect documentation on the Internet. This is 100% doing nothing and will ONLY execute 'onedrive --synchronize' .. Stop following bad Internet documentation and only refer to https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md to configure the client
```

**Reply by BKSVitor** [reply]
Abraunegg, thanks for the suggestions. I fixed the script and it continues to work well even after the changes.

But I have some doubts about the upload issue, do I need to open a new discussion or can you help me right here in this discussion?

**Reply by abraunegg** [reply]
@BKSVitor 
> But I have some doubts about the upload issue, do I need to open a new discussion or can you help me right here in this discussion

What upload issue? You had a disk I/O error that was caused by:
* Running 2 instances
* Disk being full

Nothing has been indicated about an upload issue this far.

Steps for you to take:
* Stop reading all other Internet pages for this client - they are incorrect.
  * Your source of truth are the documents listed here: https://github.com/abraunegg/onedrive/tree/master/docs
* Install the client the *correct* way for your platform. If you use Ubuntu, you *must* follow these instructions: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md


Please open a new issue once you have followed the points above.
